### PR TITLE
Refactor repo unit "disabled" check

### PR DESCRIPTION
**Pull request by wxiaoguang**
1. There are already global "unit consts", no need to use context data, which is fragile
2. Remove the "String()" method from "unit", it would only cause rendering problems in templates

### CODE REVIEW

File: routers/web/web.go [hunk 1]
```diff
@@ -487,7 +487,7 @@ func registerRoutes(m *web.Route) {
 		m.Get("/organizations", explore.Organizations)
 		m.Get("/code", func(ctx *context.Context) {
 			if unit.TypeCode.UnitGlobalDisabled() {
-				ctx.NotFound(unit.TypeCode.String(), nil)
+				ctx.NotFound("Rep unit code is disabled", nil)
```
Comment (silverwind) [inline routers/web/web.go]:
```suggestion
				ctx.NotFound("Repo unit code is disabled", nil)
```

File: routers/web/web.go [hunk 2]
```diff
@@ -384,18 +384,18 @@ func registerRoutes(m *web.Route) {
 		return func(ctx *context.Context) {
 			// only check global disabled units when ignoreGlobal is false
 			if !ignoreGlobal && unitType.UnitGlobalDisabled() {
-				ctx.NotFound(unitType.String(), nil)
+				ctx.NotFound("Repo unit is is disabled:"+unitType.LogString(), nil)
```
Comment (silverwind) [inline routers/web/web.go]:
```suggestion
				ctx.NotFound("Repo unit is is disabled: "+unitType.LogString(), nil)
```

File: templates/repo/issue/view_content/comments.tmpl [hunk 3]
```diff
@@ -574,7 +574,6 @@
 				{{template "repo/commits_list_small" dict "comment" . "root" $}}
 			{{end}}
 		{{else if eq .Type 30}}
-			{{if not $.UnitProjectsGlobalDisabled}}
```
Comment (silverwind) [inline templates/repo/issue/view_content/comments.tmpl:577]: Condition no longer needed?

Comment (wxiaoguang) [reply templates/repo/issue/view_content/comments.tmpl:577]: I do not think the condition makes sense. No matter the project is disabled or not, the issue activities should be displayed as-is.

If the condition would be considered as needed, please help to show a real case for it.

Comment (silverwind) [reply templates/repo/issue/view_content/comments.tmpl:577]: Seems fine. I agree it does not make sense to hide it.

**Review by silverwind — APPROVED** [review]

**Review by yp05327 — APPROVED** [review]

**Pull request merged**